src/mahoji/lib/bingo.ts

```diff
@@ -40,256 +53,231 @@ type BingoTile = (
+		]
 	},
 	{
 		id: 14,
-		name: 'Obtain one of: Phoenix, Tiny tempor, Youngllef, Smolcano',
-		oneOf: resolveItems(['Phoenix', 'Tiny tempor', 'Youngllef', 'Smolcano'])
+		name: 'Receive all GWD uniques (excluding pets)',
+		oneOf: [
+			'Godsword shard 1',
+			'Godsword shard 2',
+			'Godsword shard 3',
+
+			'Armadyl crossbow',
+			'Saradomin hilt',
+			'Saradomin sword',
+			"Saradomin's light",
+
+			'Bandos chestplate',
+			'Bandos tassets',
+			'Bandos boots',
+			'Bandos hilt',
+
+			'Armadyl helmet',
+			'Armadyl chestplate',
+			'Armadyl chainskirt',
+			'Armadyl hilt',
+
+			'Staff of the dead',
+			'Zamorakian spear',
+			'Steam battlestaff',
+			'Zamorak hilt'
+		]
 	},
 	{
 		id: 15,
-		name: 'Receive 2 unique godsword hilts',
-		customReq(cl) {
-			return (
-				resolveItems(['Ancient hilt', 'Armadyl hilt', 'Bandos hilt', 'Saradomin hilt', 'Zamorak hilt']).filter(
-					i => cl.has(i)
-				).length >= 2
-			);
-		}
+		name: 'Receive a Fishing Trophy from Sea Kraken',
+		allOf: ['Fishing trophy']
 	},
 	{
 		id: 16,
-		name: 'Receive a black tourmaline core',
-		allOf: resolveItems(['Black tourmaline core'])
+		name: 'Receive all 3 Dagannoth Kings pets',
+		allOf: ['Pet dagannoth prime', 'Pet dagannoth supreme', 'Pet dagannoth rex']
 	},
 	{
 		id: 17,
-		name: 'Receive a Tome of water or Tome of fire',
-		oneOf: resolveItems(['Tome of fire', 'Tome of water (empty)'])
+		name: 'Receive 3 unique nex armor pieces for a certain armor slot',
+		oneOf: [
+			['Torva full helm (broken)', 'Pernix cowl (broken)', 'Virtus mask (broken)'],
+			['Torva platebody (broken)', 'Pernix body (broken)', 'Virtus robe top (broken)'],
+			['Torva platelegs (broken)', 'Pernix chaps (broken)', 'Virtus robe legs (broken)'],
+			['Torva boots (broken)', 'Pernix boots (broken)', 'Virtus boots (broken)'],
+			['Torva gloves (broken)', 'Pernix gloves (broken)', 'Virtus gloves (broken)']
+		]
 	},
 	{
 		id: 18,
-		name: 'Receive any Champion scroll',
-		oneOf: resolveItems(championScrolls)
+		name: 'Receive any Nihiliz unique',
+		oneOf: ['Nihil horn', 'Zaryte vambraces', 'Nexling']
 	},
 	// Row 4
 	{
 		id: 19,
-		name: 'Receive/chop 5000 mahogany logs',
-		customReq(cl) {
-			return cl.amount('Mahogany logs') >= 5000;
-		}
+		name: 'Receive all Tangleroot variants from Treebeard',
+		oneOf: TanglerootTable.allItems
 	},
 	{
 		id: 20,
-		name: 'Receive a Draconic visage',
-		allOf: resolveItems(['Draconic visage'])
+		name: 'Receive 5 Gorajan shards',
+		receiveBank: new Bank().add('Gorajan shards', 5)
 	},
 	{
 		id: 21,
-		name: 'Receive a Dragon pickaxe',
-		allOf: resolveItems(['Dragon pickaxe'])
+		name: 'Receive a unique from Baxtorian Bathhouses',
+		oneOf: ['Phoenix eggling', 'Inferno adze', 'Flame gloves', 'Ring of fire']
 	},
 	{
 		id: 22,
-		name: 'Receive a Basilisk jaw',
-		allOf: resolveItems(['Basilisk jaw'])
+		name: 'Receive a unique from Monkey Rumble',
+		allOf: resolveItems(['Marimbo statue', 'Monkey egg', 'Big banana'])
```

Comment:
allOf > oneOf

This one is tricky, Monkey egg and monkey dye can be got from chimpling. Monkey crate can be bought, or got on the base table 1/25. The only uniques left are big banana + Statue which is super rare

Could this task be Defeat a special monkey in Monkey Rumble? Otherwise people can stack chimpling or there's big chance of getting purple and your item being excluded

Reply:
I think that’s fair, if its easy to code :)